src/common/include/vector/value_vector.h

```diff
@@ -41,30 +37,46 @@ class ValueVector {
         return nullMask->hasNoNullsGuarantee();
     }
 
-    inline void setRangeNonNull(uint64_t startPos, uint64_t len) {
+    inline void setRangeNonNull(uint32_t startPos, uint32_t len) {
         for (auto i = 0u; i < len; ++i) {
             setNull(startPos + i, false);
         }
     }
 
     inline uint64_t* getNullMaskData() { return nullMask->getData(); }
 
-    inline void setNull(uint64_t pos, bool isNull) { nullMask->setNull(pos, isNull); }
+    inline void setNull(uint32_t pos, bool isNull) { nullMask->setNull(pos, isNull); }
 
     inline uint8_t isNull(uint32_t pos) const { return nullMask->isNull(pos); }
 
     inline uint32_t getNumBytesPerValue() const { return numBytesPerValue; }
 
-    inline node_offset_t readNodeOffset(uint64_t pos) const {
+    template<typename T>
+    T getValue(uint32_t pos) const {
+        return ((T*)valueBuffer.get())[pos];
+    }
+
+    // Note: this function doesn't handle overflow for STRING and LIST.
+    template<typename T>
+    void setValue(uint32_t pos, T val) {
```

Comment:
I would special template for ku_list and ku_string 

```
template<>
void setValue(uint32_t pos, ku_string val) {
     addString(pos, val)
}

```

Reply:
I think you mean string here. I add a specialized template for it.
For ku_list and ku_string, we expect their overflows are already properly populated inside the vector, and if not, they actually should go through ValueVectorUtils, instead of this interface.